Update to JMH 1.10

sergiotudela opened the pull request:
* Updated to JMH 1.10
* Added the missing samples from JMH official page
* Fixed wrong link from README file to point the scala samples

ktoso commented:
Hi and thanks for the PR! 
Seems compilation failed for some weird reason?

```
[info] [error] /tmp/sbt_bfaf850c/jmh-asm/src/main/scala/org/openjdk/jmh/samples/JMHSample_32_BulkWarmup.scala:42: type mismatch;
[info] [error]  found   : org.openjdk.jmh.annotations.Mode(AverageTime)
[info] [error]  required: Array[org.openjdk.jmh.annotations.Mode]
[info] [error]  @BenchmarkMode(Mode.AverageTime)
[info] [error]                      ^
```

sergiotudela commented:
The previous PR failed for two reasons:

1. ```@BenchmarkMode``` annotation is expecting an Array, my IDE (Intelilj) didn't show me nothing strange but SBT fails, so I changed  ```@BenchmarkMode(Mode.AverageTime)``` by ```@BenchmarkMode(Array(Mode.AverageTime))```

2. ```JMHSample_35_Profilers``` sample is failing with this exception:
```
[error] Annotation generator had thrown the exception.
[error] java.lang.InternalError: Malformed class name
[error] 	at java.lang.Class.getSimpleName(Class.java:1330)
[error] 	at java.lang.Class.getCanonicalName(Class.java:1399)
[error] 	at org.openjdk.jmh.generators.reflection.RFClassInfo.getQualifiedName(RFClassInfo.java:67)
[error] 	at org.openjdk.jmh.generators.core.BenchmarkGenerator.buildAnnotatedSet(BenchmarkGenerator.java:245)
[error] 	at org.openjdk.jmh.generators.core.BenchmarkGenerator.generate(BenchmarkGenerator.java:111)
[error] 	at org.openjdk.jmh.generators.bytecode.JmhBytecodeGenerator.main(JmhBytecodeGenerator.java:100)
[error] 	at pl.project13.scala.sbt.JmhPlugin$.generateBenchmarkJavaSources(JmhPlugin.scala:103)
[error] 	at pl.project13.scala.sbt.JmhPlugin$$anonfun$projectSettings$6.apply(JmhPlugin.scala:47)
[error] 	at pl.project13.scala.sbt.JmhPlugin$$anonfun$projectSettings$6.apply(JmhPlugin.scala:47)
[error] 	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
[error] 	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
[error] 	at sbt.std.Transform$$anon$4.work(System.scala:63)
[error] 	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
[error] 	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
[error] 	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
[error] 	at sbt.Execute.work(Execute.scala:235)
[error] 	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
[error] 	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
[error] 	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[error] 	at java.lang.Thread.run(Thread.java:745)
```
  This sample is using the following syntax: ```object A { object B { ... } ... }```, so I extracted ```object B {...}``` outside. Here you can see: https://github.com/sergiotudela/sbt-jmh/commit/842737886e780c632089baae2eb3a5392cb3e8c7#diff-33de5c4384faa67aeb0bbc1626b17306L87

ktoso commented:
Excellent, reviewed and looks good. Thanks a bunch :-)
